### ISSUE TITLE
Remove MeshWorker framework

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#15839]
I believe that we agreed that for the 10.0 release, we should remove most of the MeshWorker framework with the exception of `MeshWorker::loop()`. This includes also removing step-12b, step-16b, and either removing or substantially rewriting step-39.

**Comment by drwells** [comment]
Yes - this is something we agreed upon at the workshop. We've reached a point where no-one knows how this subsystem works and with lambdas, FEInterfaceValues, and `MeshWorker::loop()` we have a reasonable replacement.

If anyone is using MeshWorker to solve problems these days we should meet with them and discuss an upgrade path so they aren't left hanging.

**Comment by drwells** [comment]
If we remove MeshWorker we should also remove everything in `matrix_block.h` since it is only ever used in that subsystem. Like our old friend `PointerMatrixBase`, we now have `BlockLinearOperator` which is a general replacement for all these classes.

**Comment by tamiko** [comment]
@drwells I think the `PointerMatrix` and `PointerMatrixAux` dragons have been slain?

**Comment by nils-schween** [comment]
I would like to comment, that the Tutorial Step-12b was very useful for me (independently from the MeshWorker framework), because it explains how to build a "face_worker" without using FEInterfaceValues. I do not know about the current status of FEInterfaceValues, but it did not work with a simple FESystem as explained in Step-8. Moreover, I use an upwind flux together with a system and I could exploit some features of my system to avoid computing eigenvalues and eigenvectors at each interface, but this required to write a "customized" face_worker. 

Bottom line, if you decide to remove Step-12b, it would be great if a new tutorial would be included for a system of PDEs with an explanation of the face_worker as simple as in 12b. 

I am currently very busy, but next year, I could offer to write one.

**Comment by bangerth** [comment]
@nils-schween More tutorial programs are always welcome! I suspect that you can do most everything with `MeshWorker::loop()`, and that function is going to remain in place. It is specifically the integrators, `InfoBox`, `IntegrationBox`, etc., that we want to remove.

**Comment by nils-schween** [comment]
Thanks for your answer and, yes, I am actually only using `MeshWorker::mesh_loop()` and none of the others. For me it is perfectly fine to remove them.
I am sorry, I think my comment was unclear. In Tutorial Step-12b, there is the method `AdvectionProblem<dim>::integrate_face_term` in which the integration over the face is done without using `FEInterfaceValues`, but with the four matrices u_1v_1, u_1v_2 etc. And I think this is how you have to do it, if you cannot use `FEInterfaceValues`. I did not find another tutorial where it is explained how to write `integrate_face_term` so well and in such a simple context. I am only lamenting the loss of this explanation. 

But let me see, if I manage to find time in January, maybe I can write a tutorial about a linear advection problem, but as  a system of PDEs.

**Comment by bangerth** [comment]
#19863 gets us partway where this issue wants to take us because it will allow us to deprecate the classes mentioned in #17696.

**Comment by bangerth** [comment]
Yay! That's been on our list for a very long time already.
